src/live_restore/live_restore_fs.c

```diff
@@ -744,7 +742,7 @@ __live_restore_fh_read(
     WT_DECL_RET;
     WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh;
     WT_SESSION_IMPL *session;
-    char *read_data;
+    char *read_data, *tmp_buf = NULL;
```

Comment:
Are you assigning here so that we don't cause clang-sanitizer issues? Given this variable is only ever used in a diagnostic block it should be defined, used and freed in a diagnostic block to avoid things getting angry about it not being used.